### PR TITLE
http: pathless CONNECT

### PR DESCRIPTION
**Pull request by alyssawilk**
Removing the synthetic path added to CONNECT requests theoretically completing Envoy CONNECT support.

Additional Description:

This has the side-effect of changing Envoy's default response to CONNECT requests from 403 (upgrade forbidden) to 404 (route not found) because without the path header, CONNECT request fail to match traditional route rules.

Risk Level: Medium (high for early CONNECT users, low otherwise)
Testing: Altered tests
Docs Changes: will un-hide in a follow-up
Release Notes: added
Runtime guard: envoy.reloadable_features.stop_faking_paths
Part of #1451 and #1630

### CODE REVIEW

**Comment by alyssawilk** [comment]
Hey @mattklein123 would like your thoughts on the behavioral change.

I was going to special case, and change the HCM "send upgrade rejected" response to happen if an upgrade failed (route match, no opgrade configured) *or* for CONNECT requests which had no route match, to preserve the existing behavior.  I feel like 403 on connect is more clear than 404, and behavioral changes and runtime guards, yadda yadda.

I had second thoughts, because if we add more logic to the extensible connect matcher, it gets complicated to tell if the 404 is a connect rejection, or if a fancy extended connect matcher failed to match.  Basically keeping the old behavior doesn't seem very future proof.

Wanted to get your take on which way to go, and I'll either runtime guard and add release notes and such, or special-case to preserve existing behavior, revert the test changes, and we'll have to remember to be careful if anyone extends the connect matcher.

**Comment by alyssawilk** [comment]
Also I don't remember where we left off with our comfort with Pathless connect.  I think the plan was to improve fuzzing and I think @asraa has done that so I think we're OK, but if you want more guards in place before this lands LMK

File: source/common/http/http2/codec_impl.cc [hunk 1]
```diff
@@ -164,6 +164,9 @@ void ConnectionImpl::ClientStreamImpl::encodeHeaders(const RequestHeaderMap& hea
   } else if (headers.Method() && headers.Method()->value() == "CONNECT") {
     modified_headers = createHeaderMap<RequestHeaderMapImpl>(headers);
     modified_headers->setProtocol(Headers::get().ProtocolValues.Bytestream);
+    if (!headers.Path()) {
+      modified_headers->setPath("/");
+    }
```
Comment (mattklein123) [inline source/common/http/http2/codec_impl.cc:169]: Is this required by the spec for H2? If so can you add comments with RFC reference?

**Comment by mattklein123** [comment]
LGTM but needs a format fix.

/wait

**Pull request merged**